bool should_plan
string ref_pos
